examples/processor/reverse/reverse.go

```diff
@@ -0,0 +1,93 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+Copyright 2016 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reverse
+
+import (
+	"strconv"
+
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
+)
+
+// RProcessor test processor
+type RProcessor struct {
+}
+
+// Process test process function
+func (r RProcessor) Process(mts []plugin.Metric, cfg plugin.Config) ([]plugin.Metric, error) {
+	metrics := []plugin.Metric{}
+	for _, m := range mts {
+		switch m.Data.(type) {
+		case int:
+			m.Data = stringToInt(reverse(intToString(m.Data.(int))))
+		case int32:
+			i32 := int(m.Data.(int32))
+			m.Data = stringToInt(reverse(intToString(i32)))
+		case int64:
+			i64 := int(m.Data.(int64))
+			m.Data = stringToInt(reverse(intToString(i64)))
+		case float32:
+			f32 := float64(m.Data.(float32))
+			m.Data = stringToFloat(reverse(floatToString(f32)))
+		case float64:
+			m.Data = stringToFloat(reverse(floatToString(m.Data.(float64))))
+		case string:
+			m.Data = reverse(m.Data.(string))
+		}
+		metrics = append(metrics, m)
+	}
+	return metrics, nil
+}
+
+// GetConfigPolicy testing config
+func (r RProcessor) GetConfigPolicy() (plugin.ConfigPolicy, error) {
+	policy := plugin.NewConfigPolicy()
+
+	br, _ := plugin.NewBoolRule(
+		"testbool",
+		false)
+	policy.AddBoolRule([]string{"random"}, br)
+
+	return *policy, nil
+}
+
+func reverse(s string) string {
+	r := []rune(s)
+	for i, j := 0, len(r)-1; i < len(r)/2; i, j = i+1, j-1 {
+		r[i], r[j] = r[j], r[i]
+	}
```

Comment:
@cody, your code does not run:-) LOL. What I used actually is Andrew Gerrand's code: https://github.com/golang/example/blob/master/stringutil/reverse.go.

From what I read, string stores in []byte. It's better to convert it to rune(int32) for the efficiency.
https://blog.golang.org/strings